MAINTAINERS.md

```diff
@@ -132,67 +130,27 @@ During your review, consider the following points:
     the author is the one most familiar with the changes they are introducing, they are the party
     most likely to resolve conflicts correctly.
 
-
 ## Workflow for merging Pull Requests
 
-Check which branch the PR was made against — patches made against the `develop` branch should only
-be merged to `develop`. Patches made against the `master` branch may be merged to only `develop`, or
-both `develop` AND `master`. The criteria to use in this latter situation is: does the patch
-introduce anything new to the API? does it potentially mark a change in behavior of any sort? If the
-answer to either of these is "yes", merge only to the `develop` branch.
-
-To sum up:
-
-- If the commit is a bugfix, merge to **both** `master` **and** `develop`.
-- If the commit is a new feature, merge to `develop` **only**.
-
-### Start the process with a clean checkout
-
-Before you begin merging, do the following:
-
-```console
-$ git fetch origin
-$ git checkout master && git rebase origin/master
-$ git checkout develop && git rebase origin/develop
-```
-
-This will set each branch to the latest commit from the canonical repository.
-
-### Create a local merge branch
-
-Next, create a new branch locally for the change, based off the appropriate branch. Use the prefix
-`hotfix/` for fixes, and `feature/` for features; typically, use the PR number as the branch name.
-
-```console
-$ git checkout -b hotfix/2854 master
-$ git checkout -b feature/2719 develop
-```
+Pull Requests should only be merged into the correct release branches:
 
-### Pull in the changeset
+- If the PR is a bugfix, it should target one of the previous release branches.
+  The only exception is when the package is not yet released and therefore has only 
+  one release branch.
+- If the PR is a new feature, merge it to the latest release branch **only**.
 
-Now you're staged and ready to pull the changeset in. On the bar that occurs before comments, on the
-left side is a little "information" icon. Click on this, and you get a popup dialog showing the
-steps to take to manually merge. If you click the icon next to the second step, it will copy a "git
-pull" command to your clipboard. Paste this into your terminal.
+You may merge it by using the green "Merge" button in github. If you do so, use only the
+"Create a merge commit" option. Do **not** use "Squash and merge" or "Rebase and merge". 
```

Comment:
I don't think we we ever discussed this in the TSC. What's the rationale for "create a merge commit" as the selected option? "Rebase and merge" leads to a more linear tree, which makes granular reversions of changes easier, while "Squash and merge" makes it easy to back out the entire changeset at once (though I'd argue squash and merge is _never_ a good idea because you lose granularlity). I've often used the `merge --ff` option when manually merging, which is similar to "Rebase and merge".

Maybe @Ocramius could weigh in?

Reply:
Actually this is just a suggestion since there is official statement here and I tried to base this on the previous gitflow strategy. But that all is a matter of taste and whatever is deemed to be important about the commit history. 

- A merge commit makes the origin of a change more obvious in the commit graph IMHO. 
- A squashed commit will consider the PR as a single commit which is sometimes useful to get rid of small adjustment commits during review rounds. It requires discipline to create small changesets though (which is always encouraged from a maintainers pov) - but this might drive off some contributors that are not familiar with this process.
- With a rebase and merge you'd loose the ability to track commits for its origin, but it produces a linear commit history.

I could also rephrase it to a "Pick whatever suits the individual situation for the PR" - which might be the best option here.